src/pages/ProposalDetailPage.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect } from 'react';
+import { OidcSecure } from '@axa-fr/react-oidc';
+import { useParams } from 'react-router-dom';
+import { usePdcApi } from '../use-pdc-api';
+import { ProposalDetailPanel } from '../components/ProposalDetailPanel';
+
+interface CanonicalField {
+  id: number;
+  label: string;
+  shortCode: string;
+}
+
+interface Proposal {
+  versions: {
+    version: number;
+    fieldValues: {
+      applicationFormField: {
+        canonicalFieldId: number;
+        position: number;
+      };
+      value: string;
+    }[];
+  }[];
+}
+
+const getValueOfCanonicalField = (
+  canonicalFields: CanonicalField[],
+  proposal: Proposal,
+  canonicalFieldShortCode: string,
+) => {
+  const field = canonicalFields.find(({ shortCode }) => (
+    shortCode === canonicalFieldShortCode
+  ));
+  if (field === undefined) {
+    return undefined;
+  }
+  const fieldValue = proposal.versions[0].fieldValues.find(({ applicationFormField }) => (
+    applicationFormField.canonicalFieldId === field.id
+  ));
+  return fieldValue?.value ?? undefined;
+};
+
+const mapCanonicalFields = (
+  canonicalFields: CanonicalField[],
+  proposal: Proposal,
+) => (
+  proposal.versions[0].fieldValues.map(({ applicationFormField, value }) => ({
+    fieldName: canonicalFields.find(({ id }) => (
+      id === applicationFormField.canonicalFieldId
+    ))?.label ?? 'missing',
+    position: applicationFormField.position,
+    value,
+  }))
+);
+
+const getApplicant = (
+  canonicalFields: CanonicalField[],
+  proposal: Proposal,
+) => (
+  getValueOfCanonicalField(canonicalFields, proposal, 'organization_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'organization_dba_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'organization_legal_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_primary_contact_name')
+    ?? getValueOfCanonicalField(canonicalFields, proposal, 'proposal_submitter_name')
+    ?? 'Unknown Applicant'
+);
+
+const ProposalDetailPage = () => {
+  const params = useParams();
+  const proposalId = params.proposalId ?? 'missing';
+  const canonicalFields = usePdcApi<CanonicalField[]>('/canonicalFields');
+  const proposal = usePdcApi<Proposal>(`/proposals/${proposalId}?includeFieldsAndValues=true`);
+
+  useEffect(() => {
+    if (canonicalFields === null || proposal === null) {
+      document.title = 'Proposal Detail - Philanthropy Data Commons';
+    } else {
+      const applicant = getApplicant(canonicalFields, proposal);
+      document.title = `${applicant} Proposal Detail - Philanthropy Data Commons`;
+    }
+  }, [canonicalFields, proposal]);
```

Comment:
When you navigate away from a proposal detail page, the title remains as set here.

Ideally, all our pages would define their titles, so this would be immediately overwritten. But we might also consider returning a cleanup function from this hook which resets the title to some default (e.g., "Philanthropy Data Commons") in case the destination page _doesn't_ have a title defined.

The downside to this is (1) the potential for the title to change twice on every navigation away slowly enough for people to notice and (2) the added boilerplate for every page that uses a title. I suspect, however, we'll want to abstract out some of this title-setting anyway.

For here and now, maybe just do the cleanup method? I think that would just mean adding `return () => document.title = 'Philanthropy Data Commons';` to the end of the hook.

Reply:
This is a fantastic note - thank you for catching it!

I will do.